politeiawww/politeiawww.go

```diff
@@ -858,6 +866,92 @@ func (p *politeiawww) handleWebsocketWrite(wc *wsContext) {
 	}
 }
 
+// updateBestBlock updates the cached best block.
+func (p *politeiawww) updateBestBlock(bestBlock uint64) {
+	p.bbMtx.Lock()
+	defer p.bbMtx.Unlock()
+	p.bestBlock = bestBlock
+}
+
+// getBestBlock returns the cached best block if there is an active
+// websocket connection to dcrdata. Otherwise it makes a call to dcrdata to
+// try to fetch the value.
+func (p *politeiawww) getBestBlock() (uint64, error) {
+	if !p.subscibedToBestBlock {
+		return p.getBestBlockDecredPlugin()
+	}
+
+	p.bbMtx.RLock()
+	defer p.bbMtx.RUnlock()
+	return p.bestBlock, nil
+}
+
+// reconnectBestBlockSub stops the connection to dcrdata, starts a new one,
+// and subscribes to the new block event.
+func (p *politeiawww) reconnectNewBlockSub() {
+	if p.wsDcrdata != nil {
+		p.wsDcrdata.client.Stop()
+		p.wsDcrdata = nil
+	}
+
+	var err error
+	p.subscibedToBestBlock = false
+
+	for {
+		p.wsDcrdata, err = newWSDcrdata()
+		if err != nil {
+			log.Errorf("reconnectWS error: %v", err)
+		}
+
+		if p.wsDcrdata != nil {
+			err = p.setupNewBlockSub()
+			if err == nil {
+				log.Debugf("Successfully reconnected to new block events")
+				break
+			}
+		}
+
+		log.Errorf("Retrying ws dcrdata reconnect in 1 minute...")
+		time.Sleep(1 * time.Minute)
+	}
+}
+
+// setupNewBlockSub handles new block and hang up messages from dcrdata.
+func (p *politeiawww) setupNewBlockSub() error {
+	err := p.wsDcrdata.subToNewBlock()
+	if err != nil {
+		return err
+	}
+
+	p.subscibedToBestBlock = true
+
+	go func() {
+		for {
+			msg, ok := <-p.wsDcrdata.client.Receive()
+			if !ok {
+				break
+			}
+			if msg == nil {
+				log.Errorf("ReceiveMsg failed")
+				continue
+			}
+
+			switch m := msg.Message.(type) {
+			case *exptypes.WebsocketBlock:
+				log.Debugf("Message WebsocketBlock(height=%s)", m.Block.Height)
+				p.updateBestBlock(uint64(m.Block.Height))
+			case *pstypes.HangUp:
```

Comment:
Glad to see this is being handled.  The pubsub server that send hangup message is deployed now.